### PR TITLE
feat: Slick card look

### PR DESCRIPTION
**Pull request by Dhruwang**
## What does this PR do?

Fixes https://github.com/formbricks/internal/issues/74

https://github.com/formbricks/formbricks/assets/67850763/2f0a67d8-617c-46b2-9cd1-6b5fd354e0a5

https://github.com/formbricks/formbricks/assets/67850763/7d9ba85c-06e5-403e-9c8f-54a5256581ee

## How should this be tested?

Go to card settings
Choose among the available card arrangements

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Review by jobenjada — CHANGES_REQUESTED** [review]
// I will do some tweaking myself and update this comment accordingly. Will ping you when it's your turn again :)

Hey @Dhruwang 

thanks for this, looks really slick already :) We'll have to do a bit of tweaking to get to the look & feel that we want for the cards and overall UX:

## Settings UX

1. Only display relevant settings:
image

We know here what survey type this survey is so pls only display the relevant setting.

-------

1.b) Text size on Settings page not matching

image

------

2. Backwards compatibility: My existing survey was rendered with "casual".
image

I think we should set all existing surveys to `straight` or `simple` (I prefer `straight` to get the look out there :) )

-------

Highlight border clashes:

image

(looks like a 🦓 )


--------

Date Picker and Cal Embed not working
image

image

----------


## Cards Look and Feel

1. The cards don't get smaller the further they are away:

image

Now that I looked for where I designed that I couldn't even find it 🙈  Pls scale them down a bit and the further they are away the smaller they get.

--------

2. Corner border weak: The border is inconsistent. Please make sure it doesnt fade out on the corners:

image

--------

3. Animation issue: You can see the content of the Thank You card while the second card fades in with opacity. 

https://github.com/formbricks/formbricks/assets/72809645/f1355430-5759-4bc4-8451-770d479592b2

----------

4. It seems like there is an additional empty card somewhere. You can see it best when the Thank You card fades in

https://github.com/formbricks/formbricks/assets/72809645/cac5f3a4-cdfc-466d-85a6-f6b1700735f9

---------

5. The hover effect is a bit too slow / delayed

-------

6. The progress bar looks broken. We should hide it on the cards which are staked behind:

image

--------

**Comment by jobenjada** [comment]
Hey Dhru! It's really coming together 😎

Here's my feedback:

1. Commented out "default" animation, but the cards still fade etc. Can we remove any kind of animation in the "simple" mode?

image

------

2. Slack: Height transition if card is on maxHeight (as discussed)

-------

3. Lock card height for animation out: Currently the card which is already submitted changes the height to match the height of the following card. It would feel more consistent and natural if the card wouldn't change height.

-------

4. Scale on hover: In the initial mock up i've added a scale on hover to the visible card. Gives the impression that the card stack "wakes up" when you hover over it.

image

Scale doesn't seem to work, any ideas why or suggestions on how to get that effect? Not suuuper essential but the micro interactions are usually appreciated :)

-------

That should be it :) 

Pls make sure to update the UI of Verify Email cards and other Link Survey Features.

Thanks a lot, looking 🔥🔥🔥

File: packages/surveys/src/components/general/Survey.tsx [hunk 1]
```diff
@@ -249,89 +257,102 @@ export const Survey = ({
...
             {showProgressBar && <ProgressBar survey={survey} questionId={questionId} />}
           </div>
         </div>
       </AutoCloseWrapper>
+    );
+  };
+
+  return (
+    <>
+      {console.log(ttc)}
```
Comment (mattinannt) [inline packages/surveys/src/components/general/Survey.tsx:348]: please remove console.log

File: packages/surveys/src/components/wrappers/StackedCardsContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      cardIndexTemp.push(survey.questions.length);
+    }
+    return cardIndexTemp;
+  }, [survey]);
+
+  const questionIdx = useMemo(() => {
+    if (currentQuestionId === "start") return survey.welcomeCard.enabled ? -1 : 0;
+    if (currentQuestionId === "end") return survey.thankYouCard.enabled ? survey.questions.length : 0;
+    return survey.questions.findIndex((question) => question.id === currentQuestionId);
+  }, [currentQuestionId, survey.welcomeCard.enabled, survey.thankYouCard.enabled]);
```
Comment (mattinannt) [inline packages/surveys/src/components/wrappers/StackedCardsContainer.tsx:47]: please check the linter warning and either extend the deps array (ideal case) or add an ignore statement

File: packages/surveys/src/components/wrappers/StackedCardsContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    const currentElement = cardRefs.current[questionIdx];
+    if (currentElement) {
+      if (resizeObserver.current) resizeObserver.current.disconnect();
+      resizeObserver.current = new ResizeObserver((entries) => {
+        for (const entry of entries) setCardHeight(entry.contentRect.height + "px");
+      });
+      resizeObserver.current.observe(currentElement);
+    }
+    return () => resizeObserver.current?.disconnect();
+  }, [questionIdx, cardRefs.current]);
```
Comment (mattinannt) [inline packages/surveys/src/components/wrappers/StackedCardsContainer.tsx:97]: please check the linter warning and either extend the deps array (ideal case) or add an ignore statement

**Review by mattinannt — COMMENTED** [review]
@Dhruwang smaller things I saw while scrolling through the code. Please fix before the merge :-)

**Comment by jobenjada** [comment]
hey @Dhruwang 

here are some more things I noticed:

1. The next button pops up in the out animation. Pls resolve:

https://github.com/formbricks/formbricks/assets/72809645/592addf3-e913-4336-a4c8-1a844bda0081

-----

2. Is this an optical illusion or do the cards move down and to the right? It's really hard to tell :D Not horrible but if you have an idea why its not moving straight down, pls check ;) 

https://github.com/formbricks/formbricks/assets/72809645/966312fc-4393-4acc-bac2-f0f1c6badb76

-------

3. if you go from a long card to a shorter one you can see the Thank You tick flickr up briefly:

https://github.com/formbricks/formbricks/assets/72809645/96154786-c921-46b3-b538-346e1fbb413f

Any non-hacky idea to solve this?

------

4. UX Remark: Is there a non-hacky way to reset the survey back to start on switching the card arrangement? If your at the end of the survey (or only have 1 cards left) the difference isn't really apparent:

image

---------

5. In app: The Close icon is acting weird. It's missing on most cards and sometimes coming through:

https://github.com/formbricks/formbricks/assets/72809645/6815e189-d67d-4cde-bbe7-033a9b0ff517

------

6. In app: Height change smoothness

The height change is much smoother for links surveys compared to in-app. Any idea why? 🤔 

Uploading slick-smoothness.mp4…

-------

7. If we choose "Simple" on In-app surveys we see a flicker between different cards:
https://github.com/formbricks/formbricks/assets/72809645/399a7fda-9eba-473e-b688-9e10664fbf52

Don't see it for link surveys. Pls look into it.

------

I think that's it ;)

**Review by jobenjada — APPROVED** [review]

**Pull request merged**